### PR TITLE
SASS 9974: Implement the new mini journey for Interest Savings

### PR DESCRIPTION
**Pull request by ansahdavid**
Linked to main ticket SASS-9960

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/test/controllers/dividendsBase/CheckOtherUkDividendsAmountControllerISpec.scala [hunk 1]
```diff
@@ -238,7 +238,7 @@ class CheckOtherUkDividendsAmountControllerISpec extends IntegrationTest with Di
     "return INTERNAL_SERVER_ERROR due to no session data" in {
       val application = GuiceApplicationBuilder()
         .in(Environment.simple(mode = Mode.Dev))
```
Comment (basotia) [inline it/test/controllers/dividendsBase/CheckOtherUkDividendsAmountControllerISpec.scala:240]: This piece of code is repeating several times, may be we can create a separate method like below or pass arguments to remove repetition. `def buildApplication(stockDividends: Boolean = true, miniJourneyEnabled: Boolean = true) = {
  GuiceApplicationBuilder()
    .in(Environment.simple(mode = Mode.Dev))
    .configure(config(stockDividends = stockDividends, miniJourneyEnabled = miniJourneyEnabled))
    .build()
}`

File: it/test/controllers/dividendsBase/OtherUkDividendsAmountBaseControllerISpec.scala [hunk 2]
```diff
@@ -30,7 +30,7 @@ class OtherUkDividendsAmountBaseControllerISpec extends IntegrationTest with Div
   val headers: Seq[(String, String)] = playSessionCookie() ++ Seq("Csrf-Token" -> "nocheck")
 
   ".show" should {
-    "direct to the original other uk dividends amount controller when 'split-dividends' is false" in {
+    "direct to the original other uk dividends amount controller when 'miniJourneyEnabled' is false" in {
```
Comment (basotia) [inline it/test/controllers/dividendsBase/OtherUkDividendsAmountBaseControllerISpec.scala:33]: same as above in all other places

File: it/test/controllers/dividendsBase/CheckOtherUkDividendsAmountControllerISpec.scala (hunk 1 above)
Comment (ansahdavid) [reply it/test/controllers/dividendsBase/CheckOtherUkDividendsAmountControllerISpec.scala]: I have made an extended version of this method in the next commit to allow each config argument to be passed in depending on test requirements.

File: it/test/controllers/dividendsBase/OtherUkDividendsAmountBaseControllerISpec.scala (hunk 2 above)
Comment (ansahdavid) [reply it/test/controllers/dividendsBase/OtherUkDividendsAmountBaseControllerISpec.scala:31]: This has been updated in the next commit

**Review by basotia — APPROVED** [review]

File: app/controllers/savingsBase/InterestSecuritiesCyaBaseController.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.savingsBase
+
+import com.google.inject.Inject
+import config._
+import controllers.savings.{InterestSecuritiesCYAController, SavingsInterestAmountController}
+import controllers.savingsSplit.{InterestSecuritiesCyaSplitController, SavingsInterestAmountSplitController}
```
Comment (PraneethN-Kotaru) [inline app/controllers/savingsBase/InterestSecuritiesCyaBaseController.scala:22]: are the SavingsInterestAmountController's unused?

Comment (ansahdavid) [reply app/controllers/savingsBase/InterestSecuritiesCyaBaseController.scala:22]: Thanks for pointing this out Praneeth, I have removed them in the next commit

**Review by PraneethN-Kotaru — APPROVED** [review]

**Review by Shihabur-Atos — APPROVED** [review]
LGTM

**Pull request merged**